commands/operator-sdk/cmd/add/api.go

```diff
@@ -75,67 +75,92 @@ func apiRun(cmd *cobra.Command, args []string) {
 	// scaffold pkg/apis/<group>/<version> directory
 	if err := os.MkdirAll(pkgApisDir, cmdutil.DefaultDirFileMode); err != nil {
 		log.Fatalf("failed to create %v: %v", pkgApisDir, err)
 	}
 
-	// scaffold pkg/apis/addtoscheme_<group>_<kind>.go
-	fileName := "addtoscheme_" + r.Group + "_" + r.Resource + ".go"
+	// scaffold pkg/apis/addtoscheme_<group>_<version>.go
+	fileName := "addtoscheme_" + r.Group + "_" + r.Version + ".go"
 	filePath := filepath.Join(fullProjectPath, "pkg", "apis", fileName)
 	codeGen := scaffold.NewAddToSchemeCodegen(&scaffold.AddToSchemeInput{ProjectPath: projectPath, Resource: r})
 	buf := &bytes.Buffer{}
 	if err = codeGen.Render(buf); err != nil {
 		log.Fatalf("failed to render the template for (%v): %v", filePath, err)
 	}
-	if writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
+	if err = writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
-	// scaffold pkg/apis/<group>/<version>/types.go
-	filePath = filepath.Join(pkgApisDir, "types.go")
+	// scaffold pkg/apis/<group>/<version>/<kind>_types.go
+	filePath = filepath.Join(pkgApisDir, typesFileName)
 	codeGen = scaffold.NewTypesCodegen(&scaffold.TypesInput{ProjectPath: projectPath, Resource: r})
 	buf = &bytes.Buffer{}
-	err = codeGen.Render(buf)
-	if err != nil {
+	if err = codeGen.Render(buf); err != nil {
 		log.Fatalf("failed to render the template for (%v): %v", filePath, err)
 	}
-	if writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
+	if err = writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
 	// scaffold pkg/apis/<group>/<version>/register.go
 	filePath = filepath.Join(pkgApisDir, "register.go")
 	codeGen = scaffold.NewRegisterCodegen(&scaffold.RegisterInput{ProjectPath: projectPath, Resource: r})
 	buf = &bytes.Buffer{}
-	err = codeGen.Render(buf)
-	if err != nil {
+	if err = codeGen.Render(buf); err != nil {
 		log.Fatalf("failed to render the template for (%v): %v", filePath, err)
 	}
-	if writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
+	if err = writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
 	// scaffold pkg/apis/<group>/<version>/doc.go
 	filePath = filepath.Join(pkgApisDir, "doc.go")
 	codeGen = scaffold.NewDocCodegen(&scaffold.DocInput{ProjectPath: projectPath, Resource: r})
 	buf = &bytes.Buffer{}
-	err = codeGen.Render(buf)
-	if err != nil {
+	if err = codeGen.Render(buf); err != nil {
 		log.Fatalf("failed to render the template for (%v): %v", filePath, err)
 	}
-	if writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
+	if err = writeFileAndPrint(filePath, buf.Bytes(), cmdutil.DefaultFileMode); err != nil {
```

Comment:
if we ever remove the above code that defines the`err := ...`, then all the `if err = ...` statements break. I'd prefer we use `if err := ...` to avoid that potential issue.